### ISSUE TITLE
Automatically update data files from simc git

### ISSUE THREAD

**Issue by Niphyr** [MechanicalPriest/SimcProfileParser#10]
Automatically update various required data files from simc git used to process item and spelldata.

**Comment by Niphyr** [comment]
**High Priority** - Objects and arrays from generated files:

- [ ] `static spell_data_t __spell_data[16909] = {` from `sc_spell_data.inc`. Defined in `spell_data.hpp`
- [ ] `static spelleffect_data_t __spelleffect_data[24764] = {` from `sc_spell_data.inc`. Defined in `spell_data.hpp`
- [ ] `static const std::array<dbc_item_data_t, 68855> __item_data { {` from `item_data.inc`. Defined in `item_data.hpp`
- [ ] `static const std::array<random_prop_data_t, 1300> __rand_prop_points_data { {` from `random_prop_points.inc`. Defined in `rand_prop_points.hpp`
- [ ] `static constexpr double __combat_ratings_mult_by_ilvl[][1300] = {` from `sc_scale_data.inc`. All of the arrays, preferably separated by their id [0], [1] etc.
- [ ] `static constexpr double __stamina_mult_by_ilvl[][1300] = {` from `sc_scale_data.inc`. All of the arrays, preferably separated by their id [0], [1] etc.

**Low priority** - Enumerations from code files:
- [ ] `stat_e` from `sc_enums.hpp`
- [ ] `item_mod_type` from `data_enums.hh`
- [ ] `item_class` from `data_enums.hh`
- [ ] `item_subclass_weapon` from `data_enums.hh`
- [ ] `item_subclass_armor` from `data_enums.hh`
- [ ] `item_subclass_consumable` from `data_enums.hh`
- [ ] `inventory_type` from `data_enums.hh`
- [ ] `item_quality` from `data_enums.hh`

**Comment by Niphyr** [comment]
The `.inc` files for the high priorty items are all here:

https://github.com/simulationcraft/simc/tree/shadowlands/engine/dbc/generated

The low priority enumerations:

https://github.com/simulationcraft/simc/blob/shadowlands/engine/sc_enums.hpp
https://github.com/simulationcraft/simc/blob/shadowlands/engine/dbc/data_enums.hh

**Comment by Niphyr** [comment]
Rough process:

1. Download the raw files from git
2. Parse the raw files creating a c# object tree
3. Serialise the object tree to json and save to file
4. Load the file as needed inside the application

If performance is an issue, some ideas:

1. Cache the file contents in memory after loading it the first time
2. Split the items data up based on usage (as per [this wiki page](https://github.com/MechanicalPriest/SimcProfileParser/wiki/ItemData)
3. Ensure any loaded data is disposed of after the final objects are returned from the library

> oldest legion itemid was 157759. oldest bfa itemid was 175010. oldest SL at the moment is 184330.

**Comment by Niphyr** [comment]
Added needing `item_bonus_type` from data_enums.hh

```c++
enum item_bonus_type
{
  ITEM_BONUS_ILEVEL    = 1,
  ITEM_BONUS_MOD       = 2,
  ITEM_BONUS_QUALITY   = 3,
  ITEM_BONUS_DESC      = 4,
  ITEM_BONUS_SUFFIX    = 5,
  ITEM_BONUS_SOCKET    = 6,
  ITEM_BONUS_REQ_LEVEL = 8,
  ITEM_BONUS_SCALING   = 11, // Scaling based on ScalingStatDistribution.db2
  ITEM_BONUS_SCALING_2 = 13, // Scaling based on ScalingStatDistribution.db2
  ITEM_BONUS_SET_ILEVEL= 14,
  ITEM_BONUS_ADD_RANK  = 17, // Add artifact power rank to a specific trait
  ITEM_BONUS_ADD_ITEM_EFFECT = 23,
  ITEM_BONUS_MOD_ITEM_STAT = 25, // Modify item stat to type
};
```

**Comment by Niphyr** [comment]
Added needing `item_socket_color` from data_enums.hh

**Comment by Niphyr** [comment]
Moved enumeration population to #11

**Comment by Niphyr** [comment]
Closed with #22
